t/plugin/elasticsearch-logger.t

```diff
@@ -0,0 +1,182 @@
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 2: Basic configuration verification
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.elasticsearch-logger")
+            local ok, err = plugin.check_schema({user = "elastic",
+                                                 password = "elastic",
+                                                 index = "apisix-elasticsearch-log",
+                                                 endpoint_addr = "http://127.0.0.1:10420/elasticsearch-logger/test"
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: auth configure undefined
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.elasticsearch-logger")
+            local ok, err = plugin.check_schema({user = "elastic",
+                                                 password = "elastic",
+                                                 index = "apisix-elasticsearch-log",
+                                                 })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+property "endpoint_addr" is required
+
+
+
+=== TEST 4: add plugin on routes
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "elasticsearch-logger": {
+                                "user": "elastic",
+                                "password": "elastic",
+                                "index": "apisix-elasticsearch-log",
+                                "endpoint_addr": "http://127.0.0.1:10420/elasticsearch-logger/test",
+                                "batch_max_size":1,
+                                "inactive_timeout":1
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:5601": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/opentracing"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 5: access local server
+--- request
+GET /opentracing
+--- response_body
+opentracing
+--- error_log
+elasticsearch body:
```

Comment:
We need to check the upload data in the body

Reply:
Actually,the data in body is unused for this plugin. for clickhouse-logger it print the SQL statement, for this plugin ,mybe the path is valuable to check ,and i do not know how to print it.

Reply:
Let's check the log of the request dumped by the dummy HTTP server.
Reference:
https://github.com/apache/apisix/blob/e6153b6c36a8b599fe724197b5c9137cacf5a588/t/plugin/loggly.t#L652-L657